src/vmm/src/config/mod.rs

```diff
@@ -153,16 +188,16 @@ impl TryFrom<String> for KernelConfig {
     }
 }
 /// Network device configuration.
-#[derive(Clone, Debug, Default, PartialEq)]
+#[derive(Clone, Debug, PartialEq)]
 pub struct NetConfig {
     /// Name of tap device.
     pub tap_name: String,
 }
 
-impl TryFrom<String> for NetConfig {
+impl TryFrom<&str> for NetConfig {
```

Comment:
Using `&str` instead of `String` is a really nice change, good catch! It would be nice to have it as a separate commit since it is not directly related to the Builder. If this is too troublesome, there is no problem. We generally try to keep one change per commit because this makes it easier to do reverts when they're necessary. It is also easier to write the changelog because you do not have to look at each commit in depth to see if there are additional changes that the issue it resolves.

Reply:
Yes, will do that - I was thinking of that already because regardless of whether `builder` is to be used, this is a good change. So may be I will first make that change and then look at other changes.